### PR TITLE
Add SetTextureSpace api for TextureBaker

### PR DESCRIPTION
**Pull request by ashwinbhat**
Update #921

Allows setting a uvMin and uvMax range for the TextureBaker and allows the texture space quad to be configurable.

Updates to testsuite to improve Wedge and Baking test using new nodedefs

### CODE REVIEW

File: source/MaterialXRenderGlsl/GlslRenderer.cpp [hunk 1]
```diff
@@ -138,13 +138,13 @@ void GlslRenderer::createProgram(const StageMap& stages)
     _program->build();
 }
 
-void GlslRenderer::renderTextureSpace()
+void GlslRenderer::renderTextureSpace(Vector2 uvMin, Vector2 uvMax)
```
Comment (bernardkwok) [inline source/MaterialXRenderGlsl/GlslRenderer.cpp:141]: Nitpick. Should be "const Vector2&" args.

File: source/MaterialXRenderGlsl/TextureBaker.h [hunk 2]
```diff
@@ -187,6 +187,19 @@ class TextureBaker : public GlslRenderer
         return _hashImageNames;
     }
 
+    /// Set the min and max UV values for the baking texture space
```
Comment (bernardkwok) [inline source/MaterialXRenderGlsl/TextureBaker.h:190]: Maybe: "for the geometry used for baking." instead ?

File: source/MaterialXTest/MaterialXGenShader/GenShaderUtil.h [hunk 3]
```diff
@@ -141,16 +141,23 @@ class TestSuiteOptions
...
 
     // Wedge parameters
-    mx::StringVec wedgeFiles;
-    mx::StringVec wedgeParameters;
-    mx::FloatVec wedgeRangeMin;
-    mx::FloatVec wedgeRangeMax;
-    mx::IntVec wedgeSteps;
-
+    struct wedgesetting {
+        std::string wedgefile = "";
```
Comment (bernardkwok) [inline source/MaterialXTest/MaterialXGenShader/GenShaderUtil.h:145]: Nitpick: Use EMPTY_STRING vs "".

File: source/MaterialXTest/MaterialXGenShader/GenShaderUtil.h [hunk 4]
```diff
@@ -141,16 +141,23 @@ class TestSuiteOptions
     mx::FileSearchPath externalTestPaths;
 
     // Wedge parameters
-    mx::StringVec wedgeFiles;
-    mx::StringVec wedgeParameters;
-    mx::FloatVec wedgeRangeMin;
-    mx::FloatVec wedgeRangeMax;
-    mx::IntVec wedgeSteps;
-
+    struct wedgesetting {
```
Comment (bernardkwok) [inline source/MaterialXTest/MaterialXGenShader/GenShaderUtil.h:144]: Nitpick. use camelCase so it's "wedgeSetting, wedgeSettings, bakeSetting, bakeSettings" respectively.

**Review by bernardkwok — APPROVED** [review]
Looks good. Some minor nitpicks, but signature should use const args.

File: source/MaterialXTest/MaterialXGenShader/GenShaderUtil.h (hunk 4 above)
Comment (niklasharrysson) [reply source/MaterialXTest/MaterialXGenShader/GenShaderUtil.h:144]: More nickpick: for a struct or class declaration it should be WedgeSetting, BakeSetting, etc. , to follow our normal coding convention.

**Review by bernardkwok — APPROVED** [review]
Looks good.

**Pull request merged**